hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -88,6 +89,8 @@ func (r *HostedClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Watches(&source.Kind{Type: &capiawsv1.AWSCluster{}}, handler.EnqueueRequestsFromMapFunc(enqueueParentHostedCluster)).
 		Watches(&source.Kind{Type: &hyperv1.HostedControlPlane{}}, handler.EnqueueRequestsFromMapFunc(enqueueParentHostedCluster)).
 		Watches(&source.Kind{Type: &capiv1.Cluster{}}, handler.EnqueueRequestsFromMapFunc(enqueueParentHostedCluster)).
+		Watches(&source.Kind{Type: &routev1.Route{}}, handler.EnqueueRequestsFromMapFunc(enqueueParentHostedCluster)).
```

Comment:
nit: having a comment here indicating why we watch the resources that might not be that intuitive e.g Route/Deployment would be helpful when looking at this for people with no context or even ourselves in a while from now.

Reply:
I added a blurb here; it's a little tricky because really these types alone aren't enough to map to the specific resources we reconcile and are watching for (e.g. multiple Deployments for different purposes).